File: clang/test/PCH/suspicious-pragma-pack.c
// RUN: %clang_cc1 -triple x86_64-apple-darwin10 %s -verify -emit-pch -o %t
// RUN: %clang_cc1 -triple x86_64-apple-darwin10 %s -verify -include-pch %t

#ifndef HEADER
#define HEADER
#pragma pack (push, 1)
#else
#pragma pack (2)
#endif
// expected-warning@-4 {{unterminated '#pragma pack (push, ...)' at end of file}}

File: clang/test/SemaObjC/suspicious-pragma-pack.m
// RUN: %clang_cc1 -triple i686-apple-darwin9 -fsyntax-only -I%S/Inputs -verify %s

#pragma pack (push, 1) // expected-note {{previous '#pragma pack' directive that modifies alignment is here}}
#import "empty.h" // expected-warning {{non-default #pragma pack value might change the alignment of struct or union members in the included file}}

#pragma pack (pop)
